src/sinks/loki/sink.rs

```diff
@@ -158,19 +160,31 @@ pub(super) struct EventEncoder {
 
 impl EventEncoder {
     fn build_labels(&self, event: &Event) -> Vec<(String, String)> {
-        self.labels
-            .iter()
-            .filter_map(|(key_template, value_template)| {
-                if let (Ok(key), Ok(value)) = (
-                    key_template.render_string(event),
-                    value_template.render_string(event),
-                ) {
-                    Some((key, value))
+        let mut vec: Vec<(String, String)> = Vec::new();
+
+        for (key_template, value_template) in self.labels.iter() {
+            if let (Ok(key), Ok(value)) = (
+                key_template.render_string(event),
+                value_template.render_string(event),
+            ) {
+                if key.ends_with("_*") {
+                    let pkey = key.trim_end_matches('*');
+                    let output: serde_json::map::Map<String, serde_json::Value> =
+                        serde_json::from_str(value.as_str()).unwrap();
```

Comment:
This should not `unwrap`. That will panic vector if the value turns out to not be an object.

I think if it can't be parsed as a map, we should treat it the same as if `value_template.render_string` returned an error and not include it as a label.

Reply:
You are right. I replaced this method with the `unwarp_and_default` and added a test that vector does not panic on this call.